common/src/ringbuffer.c

```diff
@@ -19,79 +19,68 @@
  */
 
 #include "common/ringbuffer.h"
-#include "common/locking.h"
+#include "common/debug.h"
+#include "common/util.h"
 
+#include <stdatomic.h>
 #include <stdlib.h>
 #include <string.h>
 
 struct RingBuffer
 {
-  RingBufferValueFn preOverwriteFn;
-  void *            preOverwriteUdata;
-
-  int     length;
-  size_t  valueSize;
-  LG_Lock lock;
-  int     start, pos, count;
-  char    values[0];
+  uint32_t length;
```

Comment:
why `uint32_t` specifically? Why not just `unsigned int` so the compiler can just select the optimal type for the platform?